public/app/features/alerting/unified/Silences.tsx

```diff
@@ -1,24 +1,32 @@
 import { Field, InfoBox, LoadingPlaceholder } from '@grafana/ui';
-import React, { FC, useEffect } from 'react';
+import React, { FC, useEffect, useMemo } from 'react';
 import { useDispatch } from 'react-redux';
 import { AlertingPageWrapper } from './components/AlertingPageWrapper';
 import { AlertManagerPicker } from './components/AlertManagerPicker';
 import { useAlertManagerSourceName } from './hooks/useAlertManagerSourceName';
 import { useUnifiedAlertingSelector } from './hooks/useUnifiedAlertingSelector';
-import { fetchSilencesAction } from './state/actions';
+import { fetchSilencesAction, fetchAllPromAndRulerRulesAction } from './state/actions';
 import { initialAsyncRequestState } from './utils/redux';
+import SilencesTable from './components/silences/SilencesTable';
+import { getRulesDataSources } from './utils/datasource';
 
 const Silences: FC = () => {
   const [alertManagerSourceName, setAlertManagerSourceName] = useAlertManagerSourceName();
   const dispatch = useDispatch();
-
+  const rulesDataSources = useMemo(getRulesDataSources, []);
   const silences = useUnifiedAlertingSelector((state) => state.silences);
+  const promRules = useUnifiedAlertingSelector((state) => state.promRules);
 
   useEffect(() => {
     dispatch(fetchSilencesAction(alertManagerSourceName));
+    dispatch(fetchAllPromAndRulerRulesAction());
```

Comment:
@gotjosh for the Silences page we fetch all the prom and ruler rules and then filter out the ones based on the matchers for that silence in a hook

Reply:
Sorry, my fault, this part will need to be redone a little bit :)

I think right now we  **should not try to figure out whichalert rules** would be matched by a given silence.
First, we don't even know which prometehus is sending to which alertmanager, and thus is affected by its silences.
Second, even if we did, I'm not sure it makes logical sense anyway, since rules don't have all the lables that an alert might have and silence might configure.

The figma label is misleading as well, my bad, it somehow slipped past me. I think the section should be "Matching alerts", and **we should get affected alerts from the same alertmanager** that we load silences from.

I mad a [PR to your PR](https://github.com/grafana/grafana/pull/33142) that adds loading alerts from the same alertmanager.

For existing silences, you can simply find alerts by checking that alerts `status.silencedBy` includes the silence id. You will need to do label matching only for alerts preview in the silence form.

You can get the rule name from the `__alert_rule_title__` label (grafana) or `alertname` label (lotex)

Sample alertmnaager alert payload below:

```
[
    {
        "annotations": {},
        "endsAt": "2021-04-20T08:56:42.139+03:00",
        "fingerprint": "57b7c1e8a48fe013",
        "receivers": [
            {
                "name": "slack_demo"
            }
        ],
        "startsAt": "2021-04-20T08:53:02.139+03:00",
        "status": {
            "inhibitedBy": [],
            "silencedBy": [
                "db813b37-3bfc-4430-973b-42249e8db22d"
            ],
            "state": "suppressed"
        },
        "updatedAt": "0001-01-01T00:00:00.000Z",
        "labels": {
            "__alert_rule_namespace_uid__": "EPO8nYEGz",
            "__alert_rule_title__": "Firing",
            "__alert_rule_uid__": "L8LYYQXMz"
        }
    }
]
```